lens.cabal

```diff
@@ -412,40 +401,17 @@ test-suite hunit
       test-framework       >= 0.6,
       test-framework-hunit >= 0.2
 
--- Verify the results of the examples
+-- We need this dummy test-suite to add simple-reflect to the install plan
```

Comment:
The fact that you need a dummy test suite for dependency purposes is interesting. Does it somehow pick up on the fact that this test suite is related to doctests?

Reply:
> Does it somehow pick up on the fact that this test suite is related to doctests?

No. It just looks around in the install plan for `--extra-package`s you specify. If `cabal.project` `extra-packages` worked, we could just use that. (They might in 3.4, but I'm not sure).

Reply:
OK. I assume this means that this will only work if `--enable-tests` is set? In other words, `docspec: True` and `tests: False` are incompatible?

Reply:
For `lens`, yes.

Reply:
OK. This all feels a bit delicate to me, but at the same time, I can't think of a better way to do this with the design considerations involved. At the very least, this use case is documented in the manual, so I'm content with that.

Reply:
I tried locally, `extra-packages: simple-reflect` seems to work with `cabal-install-3.4`.

So in near future (once it's officially released &dagger;, and support in `haskell-ci` is added, ...) this can be cleaned up a bit. I don't want go there yet though.

&dagger; because adding it to `cabal.project` will affect contributors, which may not use release candidates.